### DISCUSSION TITLE
URL, in templates, cannot reference non-handler route definitions

### DISCUSSION THREAD

**Discussion by igbanam** [martenframework/marten#243] [Q&A] [answered]
With this root routes file

```crystal
Marten.routes.draw do
  path "/nested-app", NestedApp::ROUTES, name: "nested_app"
end
```

A template file cannot reference this route using `{% url "nested_app" %}`. However, if it points to a handler, like in the example below

```crystal
Marten.routes.draw do
  path "/nested-app", NestedApp::ROUTES, name: "nested_app"
  path "/direct-to-handler", DirectHandler, name: "direct"
end
```

Then `{% url "direct" %}` properly compiles into `"/direct-to-handler"`.

I expect that with the nested routes, the same behaviour should hold?

**Comment by ellmetha** [comment, marked as answer]
Hey! 👋 

Your `nested_app` route is an [including route](https://martenframework.com/docs/handlers-and-http/routing#defining-included-routes) and as such it does not map to a concrete handler: only concrete handler routes can be resolved using the `url` template tag.

I imagine your `NestedApp::ROUTES` constant contains another route map, so instead of trying to resolve `nested_app` you should probably try to resolve the URL of one of the underlying (namespaced) routes (eg. `{% url "nested_app:sub_route" %}`).

**Reply by igbanam** [reply]
Thanks for clarifying this. I got excited about the apps separation and got stuck on the whole separation of concerns idea — i.e. an app should only know about routes defined within it. — I thought I could just point to the including route and have that redirect to a default in the included list. But now I think about it, there's no default per se.
